### PR TITLE
Optimize contracts integration

### PR DESCRIPTION
**Pull request by adibas03**
successfully fund

### CODE REVIEW

**Comment by adibas03** [comment]
@tspoff Correct,
that was the failing function.
It fails if two simultaneous calls are made to the function.
Separating the functions, seems to work

File: src/components/pages/home/hero/index.tsx [hunk 1]
```diff
@@ -96,44 +96,33 @@ class HomeHero extends React.Component<HomeHeroProps, HomeHeroState> {
...
-      const principalRequested = loanParams.principalRequested;
-      // const payees = await repaymentManagerInstance.methods._payees();
-
+      const paymentToken = await getTokenDetailsFromAddress(await getPrincipalToken(termsContractInstance));
+      
       let loanEndTimestamp;
 
-      if (loanParams.loanStartTimestamp !== "0") {
+      if (loanStartTimestamp !== "0") {
         loanEndTimestamp = await getLoanEndTimestamp(termsContractInstance);
```
Comment (djuanit0x) [inline src/components/pages/home/hero/index.tsx]: It is more clear with LoanStatuses.NOT_STARTED than "0"?

File: src/components/pages/home/hero/index.tsx [hunk 2]
```diff
@@ -96,44 +96,33 @@ class HomeHero extends React.Component<HomeHeroProps, HomeHeroState> {
...
-
-      const principalRequested = loanParams.principalRequested;
-      // const payees = await repaymentManagerInstance.methods._payees();
-
+      const paymentToken = await getTokenDetailsFromAddress(await getPrincipalToken(termsContractInstance));
+      
       let loanEndTimestamp;
 
-      if (loanParams.loanStartTimestamp !== "0") {
+      if (loanStartTimestamp !== "0") {
```
Comment (djuanit0x) [inline src/components/pages/home/hero/index.tsx]: Is it more clear with LoanStatuses.NOT_STARTED than "0"?

File: src/components/pages/home/tab/profile/index.tsx [hunk 3]
```diff
@@ -208,7 +229,7 @@ class Profile extends React.Component<{}> {
             </Margin>
           </Col>
           <Col lg={4} md="hidden">
-            <SimuLationReturn />
+            <SimuLationReturn simulateInterest={this.simulateInterest.bind(this)} />
```
Comment (djuanit0x) [inline src/components/pages/home/tab/profile/index.tsx]: Bind is redundant here

File: src/components/pages/home/hero/index.tsx (hunk 2 above)
Comment (adibas03) [reply src/components/pages/home/hero/index.tsx]: Nice catch, will update in next PR, so we can have this update available

File: src/components/pages/home/tab/profile/index.tsx (hunk 3 above)
Comment (adibas03) [reply src/components/pages/home/tab/profile/index.tsx]: will be updated in next PR

**Pull request merged**